pkg/handlers/registeredResources.go

```diff
@@ -0,0 +1,158 @@
+	_, err := h.sdk.RegisteredResources.UpdateRegisteredResource(ctx, &registeredresources.UpdateRegisteredResourceRequest{
+		Id:                     id,
+		Name:                   name,
+		Metadata:               metadata,
+		MetadataUpdateBehavior: behavior,
+	})
+	if err != nil {
+		return nil, err
+	}
+
+	return h.GetRegisteredResource(ctx, id, "")
+}
+
+func (h Handler) DeleteRegisteredResource(ctx context.Context, id string) error {
+	_, err := h.sdk.RegisteredResources.DeleteRegisteredResource(ctx, &registeredresources.DeleteRegisteredResourceRequest{
+		Id: id,
+	})
+
+	return err
+}
+
+//
+// Registered Resource Values
+//
+
+func (h Handler) CreateRegisteredResourceValue(ctx context.Context, resourceID string, value string, actionAttributeValues []*registeredresources.ActionAttributeValue, metadata *common.MetadataMutable) (*policy.RegisteredResourceValue, error) {
+	resp, err := h.sdk.RegisteredResources.CreateRegisteredResourceValue(ctx, &registeredresources.CreateRegisteredResourceValueRequest{
+		ResourceId:            resourceID,
+		Value:                 value,
+		ActionAttributeValues: actionAttributeValues,
+		Metadata:              metadata,
+	})
+	if err != nil {
+		return nil, err
+	}
+
+	return resp.GetValue(), nil
+}
+
+func (h Handler) GetRegisteredResourceValue(ctx context.Context, id, fqn string) (*policy.RegisteredResourceValue, error) {
+	req := &registeredresources.GetRegisteredResourceValueRequest{}
+	if id != "" {
+		req.Identifier = &registeredresources.GetRegisteredResourceValueRequest_Id{
+			Id: id,
+		}
+	} else {
+		req.Identifier = &registeredresources.GetRegisteredResourceValueRequest_Fqn{
+			Fqn: fqn,
+		}
+	}
+
+	resp, err := h.sdk.RegisteredResources.GetRegisteredResourceValue(ctx, req)
+	if err != nil {
+		return nil, err
+	}
+
+	return resp.GetValue(), nil
+}
+
+func (h Handler) ListRegisteredResourceValues(ctx context.Context, resourceID string, limit, offset int32) ([]*policy.RegisteredResourceValue, *policy.PageResponse, error) {
+	resp, err := h.sdk.RegisteredResources.ListRegisteredResourceValues(ctx, &registeredresources.ListRegisteredResourceValuesRequest{
+		ResourceId: resourceID,
+		Pagination: &policy.PageRequest{
+			Limit:  limit,
+			Offset: offset,
+		},
+	})
+	if err != nil {
+		return nil, nil, err
+	}
+
+	return resp.GetValues(), resp.GetPagination(), nil
+}
+
+func (h Handler) UpdateRegisteredResourceValue(ctx context.Context, id, value string, actionAttributeValues []*registeredresources.ActionAttributeValue, metadata *common.MetadataMutable, behavior common.MetadataUpdateEnum) (*policy.RegisteredResourceValue, error) {
+	_, err := h.sdk.RegisteredResources.UpdateRegisteredResourceValue(ctx, &registeredresources.UpdateRegisteredResourceValueRequest{
+		Id:                     id,
+		Value:                  value,
+		ActionAttributeValues:  actionAttributeValues,
```

Comment:
Can you please add a test to ensure the pflags implementation doesn't end up passing an empty slice instead of a nil slice to this request and therefore clearing out the stored `actionAttributeValues`?

Reply:
Good catch here. Turns out there's a bug in my platform code that will delete the existing action attribute values regardless of whether any were passed in the request:

https://github.com/opentdf/platform/blob/main/service/policy/db/registered_resources.go#L433-L437

I'll add a piece to the update test here in bats, but comment it out until the bug is fixed in platform.

Reply:
Bug is fixed in the platform and the code added to the update test is now back to uncommented and being ran